twine/package.py

```diff
@@ -18,39 +18,35 @@
 import os
 import re
 import subprocess
-import sys
-import warnings
-from typing import (
-    Any,
-    Dict,
-    Iterable,
-    List,
-    NamedTuple,
-    Optional,
-    Sequence,
-    Tuple,
-    Union,
-    cast,
-)
-
-if sys.version_info >= (3, 10):
-    import importlib.metadata as importlib_metadata
-else:
-    import importlib_metadata
-
-import packaging.version
-import pkginfo
+from typing import Any, Dict, List, NamedTuple, Optional, Tuple
+
+from packaging import metadata
+from packaging import version
 from rich import print
 
+from twine import bdist
 from twine import exceptions
+from twine import sdist
 from twine import wheel
 from twine import wininst
 
+# Monkeypatch Metadata 2.0 support
+metadata._VALID_METADATA_VERSIONS = [
+    "1.0",
+    "1.1",
+    "1.2",
+    "2.0",
+    "2.1",
+    "2.2",
+    "2.3",
+    "2.4",
+]
+
```

Comment:
Note that PyPI hasn't supported Metadata 2.0 since March of this year, and we don't plan to add it back, so this may be unnecessary: https://github.com/pypi/warehouse/issues/17168

Reply:
I know, I think I'm the one that noticed it first, see pypi/warehouse#17168 😄. However this PR does already many things. I would leave dropping support for metadata 2.0 to another conversation.

Reply:
As a reminder, twine supports more than just PyPI

Reply:
@dnicolodi Aha! Noted, and thanks for catching that :)

Reply:
Agreed @dnicolodi let's keep this PR as small as possible. But again, there are other package indexes that people use twine with that aren't PyPI. We have no way of knowing what they expect/people are expecting to use with them.